File: crates/domain-scan-core/tests/fixtures/go/interfaces.go
package main

type Reader interface {
	Read(p []byte) (n int, err error)
}

type Writer interface {
	Write(p []byte) (n int, err error)
}

type ReadWriter interface {
	Reader
	Writer
}

type UserService interface {
	GetUser(id string) (*User, error)
	CreateUser(name string, email string) (*User, error)
	DeleteUser(id string) error
}

type privateInterface interface {
	internalMethod()
}

File: crates/domain-scan-core/tests/fixtures/typescript/interfaces.ts
// Test fixture: TypeScript interfaces

export interface IUserService {
  getUser(id: string): Promise<User>;
  createUser(data: CreateUserDto): Promise<User>;
  deleteUser(id: string): Promise<void>;
}

interface ILogger {
  log(message: string, level?: string): void;
  error(message: string, error: Error): void;
}

export interface IRepository<T> {
  findById(id: string): Promise<T | null>;
  findAll(): Promise<T[]>;
  save(entity: T): Promise<T>;
  delete(id: string): Promise<boolean>;
}

export interface IConfig {
  readonly apiUrl: string;
  readonly port: number;
  debug?: boolean;
}

interface IEventEmitter extends IDisposable {
  on(event: string, handler: Function): void;
  emit(event: string, ...args: any[]): void;
}
